pkg/errcode/errcode.go

```diff
@@ -160,6 +160,69 @@ const (
 	ErrMeshConfigMarshaling
 )
 
+// Range 5000-5500 reserved for errors related to Envoy
+const (
+	// ErrMarshallingAccessLog indicates the Envoy AccessLog struct created could not be marshalled
+	ErrMarshallingAccessLog ErrCode = iota + 5000
```

Comment:
No need for a separate error code for each type when it comes to marshaling. A generic error code `ErrMarshallingXDSResource` should suffice.